backend/inspirehep/records/api/base.py

```diff
@@ -169,6 +169,20 @@ def get_record(cls, record_uuid, with_deleted=False, record_version=None):
             raise NoResultFound
         return record
 
+    @classmethod
+    def get_records_by_ids(cls, ids):
```

Comment:
how is this different from `get_records_by_pids`?

Reply:
The major difference is that in `RecordsAuthors` there are not pids, we have only ids, so unless I don't do the join in `query_author_papers` I can't get them back from db (I haven't thought about it before, but might I change it). Then the issue is that I am getting literature records by calling `InspireRecord`, so I need to get class record class [there]( https://github.com/inspirehep/inspirehep/pull/2072/files#diff-fc03de0e68b26d840b1e069376322e3b7339601165daa83a7c4ac829b547b992R183-R184). If it's ok, I can modify `get_records_by_pids` by it there

Reply:
you don't need to pass pid objects to `get_records_by_pids`, but `(pid_type, pid_value)` tuples, which is exactly what you're doing too, so I fail to see the difference. Why can't you use `LiteratureRecord` instead of `InspireRecord`, given you're already hardcoding `lit` pids?

Reply:
But id is not a pid, isn't it (and it's not in pid table)? If not, I can't pass it in the tuple because record won't be find.
No because of circular dependency issue